### ISSUE TITLE
App Hangs

### ISSUE THREAD

**Issue by JohnLBevan** [shlusiak/Freebloks-Android#2]
App hangs near end of game; when other players out of available moves.  The issue's intermittent; seems to work for the the second turn after all other players are out of moves), but often hangs after the second or third.

The game's menus continue to work, but the undo button is disabled, and the game itself is unresponsive.
You can successfully initiate a new game from the menu, "resolving" the issue (i.e. the board is no longer frozen; though the previous game is aborted).

Sometimes it's possible to also rotate the board (i.e. to see other player's remaining pieces).  On returning to your own board your pending pieces are no longer displayed.
Navigating to a different app then back again has a similar result to rotating the board full circle (i.e. your pieces are no longer displayed).

**Comment by shlusiak** [comment]
Hi John,

thank you for your detailed report, I appreciate it a lot. How easy is it for you to reproduce this issue? I haven’t seen this so far but I believe the connection to the backend is lost when waiting for the computer moves, e.g. a protocol error. When you rotate the board and back, does it show your name at the bottom or does it say “disconnected” in white writing? And you say that sometimes it freezes and you can’t even rotate the board? Can you reproduce this when having the computer play entirely against itself (custom game, uptick your color)? Can you exit the App with the back button when it hangs and does it offer you to resume the game when you start it again?

Thanks again for your help.


Sascha

**Comment by JohnLBevan** [comment]
Hey Sascha,

No worries, thank-you for the game; it`s awesome.

I see the issue once every 5 to 10 games, so quite easy to reproduce.

When rotating the board back I get “no player” in white writing.

Pressing back (in game top left back button and Android`s back button have
the same effect) I get the main menu dialog offering new game, resume, etc.
Resume takes me back to the game (still hung), other options work as
expected.
Pressing back twice exits the app (it stil displays as a running app, but
on selecting goes back to the intro animation.

I`ve not yet been able to get the error in a computer only game... Will
keep trying.

Thank-you again.

Jb

On 2 Aug 2015 07:14, "Sascha Hlusiak" <notifications@github.com> wrote:
>
> Hi John,
>
> thank you for your detailed report, I appreciate it a lot. How easy is it
for you to reproduce this issue? I haven’t seen this so far but I believe
the connection to the backend is lost when waiting for the computer moves,
e.g. a protocol error. When you rotate the board and back, does it show
your name at the bottom or does it say “disconnected” in white writing? And
you say that sometimes it freezes and you can’t even rotate the board? Can
you reproduce this when having the computer play entirely against itself
(custom game, uptick your color)? Can you exit the App with the back button
when it hangs and does it offer you to resume the game when you start it
again?
>
> Thanks again for your help.
>
>
> Sascha
>
>
>
> > On 1 Aug 2015, at 21:39, John Bevan <notifications@github.com> wrote:
> >
> > App hangs near end of game; when other players out of available moves.
The issue's intermittent; seems to work for the the second turn after all
other players are out of moves), but often hangs after the second or third.
> >
> > The game's menus continue to work, but the undo button is disabled, and
the game itself is unresponsive.
> > You can successfully initiate a new game from the menu, "resolving" the
issue (i.e. the board is no longer frozen; though the previous game is
aborted).
> >
> > Sometimes it's possible to also rotate the board (i.e. to see other
player's remaining pieces). On returning to your own board your pending
pieces are no longer displayed.
> > Navigating to a different app then back again has a similar result to
rotating the board full circle (i.e. your pieces are no longer displayed).
> >
> > —
> > Reply to this email directly or view it on GitHub <
https://github.com/shlusiak/Freebloks-Android/issues/2>.

**Comment by shlusiak** [comment]
Hi John,

I believe you are hitting an out-of-sync situation that isn’t handled well but simply accepted in the front end while the backend just swallows invalid turns, leading to a “no player” situation. I have found a few bugs that prevented from out-of-sync to be properly detected as well as a bug that would resume a game with inconsistent data. This seems to be unrelated since I don’t think a resume is involved every time, is it?

Are you able to reproduce this issue when “joining” to the default server? I have fixed some issues and it might give me more information what’s going on here.

Further I will publish an update to the App soon that will be much more strict when it comes to out-of-sync situations and in most cases simply crash, allowing me to get crash reports.


Cheers,
Sascha
